packages/dds/task-manager/src/taskManager.ts

```diff
@@ -192,7 +198,17 @@ export class TaskManager extends SharedObject<ITaskManagerEvents> implements ITa
     /**
      * Map to track tasks that have pending complete ops.
      */
-     private readonly pendingCompletedTasks: Map<string, number[]> = new Map();
+    private readonly pendingCompletedTasks: Map<string, number[]> = new Map();
+
+    /**
+     * Returns the clientId. Will return a placeholder if the runtime is detached and not yet assigned a clientId.
+     */
+    private get clientId(): string | undefined {
+        if (!this.isAttached()) {
+            return this.runtime.clientId ?? placeholderClientId;
```

Comment:
Here probably prefer just returning the placeholderClientId when detached.  This way we know there won't be stale clientIds in the attach summary (if multiple disconnect/reconnects happen while still detached).

And then replacePlaceHolderFromAllQueues can fix them up with the most-recent clientId when it does finally attach.

Maybe also just double-check whether this helper function is still useful under the new approach as compared to using this.runtime.clientId and placeholderClientId directly -- I see a number of places where we still have to double-check if we're in detached state or what value we got back, so this might not really provide much abstraction anymore.

Reply:
Makes sense. I decided to keep the getter since I still think it helps keep the logic clean in other parts of the code